models/user/user.go

```diff
@@ -891,14 +891,19 @@ func UpdateUser(ctx context.Context, u *User, changePrimaryEmail bool, cols ...s
 		if err != nil {
 			return err
 		}
-		if !has {
-			// 1. Update old primary email
-			if _, err = e.Where("uid=? AND is_primary=?", u.ID, true).Cols("is_primary").Update(&EmailAddress{
-				IsPrimary: false,
-			}); err != nil {
-				return err
+		if has && emailAddress.UID != u.ID {
```

Comment:
What if email exists and is not active? This way I could add to my account let's say others person's email address and block him from adding his other email

Reply:
Agreed, but that brings another change with it:
We need to throw out unactivated emails then when someone else changes his email and verifies it.

Reply:
It's a backport, should I modify this patch?

Reply:
Yes, I think this should be fixed